src/portfft/dispatcher/global_dispatcher.hpp

```diff
@@ -306,57 +308,94 @@ struct committed_descriptor<Scalar, Domain>::run_kernel_struct<Dir, LayoutIn, La
     for (std::size_t i = 0; i < num_batches; i += max_batches_in_l2) {
       IdxGlobal intermediate_twiddles_offset = 0;
       IdxGlobal impl_twiddle_offset = initial_impl_twiddle_offset;
+      auto& kernel0 = kernels.at(0);
       l2_events = detail::compute_level<Scalar, Domain, Dir, detail::layout::BATCH_INTERLEAVED,
                                         detail::layout::BATCH_INTERLEAVED, SubgroupSize>(
-          kernels.at(0), in, desc.scratch_ptr_1.get(), twiddles_ptr, factors_and_scan, scale_factor,
-          intermediate_twiddles_offset, impl_twiddle_offset,
-          2 * static_cast<IdxGlobal>(i) * committed_size + input_offset, committed_size,
+          kernel0, in, desc.scratch_ptr_1.get(), in_imag, desc.scratch_ptr_1.get() + imag_offset, twiddles_ptr,
+          factors_and_scan, scale_factor, intermediate_twiddles_offset, impl_twiddle_offset,
+          vec_size * static_cast<IdxGlobal>(i) * committed_size + input_offset, committed_size,
           static_cast<Idx>(max_batches_in_l2), static_cast<IdxGlobal>(num_batches), static_cast<IdxGlobal>(i), 0,
-          dimension_data.num_factors, {event}, desc.queue);
-      intermediate_twiddles_offset += 2 * kernels.at(0).batch_size * static_cast<IdxGlobal>(kernels.at(0).length);
-      impl_twiddle_offset +=
-          detail::increment_twiddle_offset(kernels.at(0).level, static_cast<Idx>(kernels.at(0).length));
-      for (std::size_t factor_num = 1; factor_num < static_cast<std::size_t>(dimension_data.num_factors);
-           factor_num++) {
-        if (static_cast<Idx>(factor_num) == dimension_data.num_factors - 1) {
+          dimension0.num_factors, storage, {event}, desc.queue);
+      std::vector<Scalar> tmp(desc.params.number_of_transforms * dimension0.length * 2);
+#ifdef PORTFFT_LOG_DUMPS
+      desc.queue.copy(desc.scratch_ptr_1.get(), tmp.data(), tmp.size(), l2_events).wait();
+      std::cout << std::endl;
+      std::cout << "after factor 0: ";
+      for (Scalar val : tmp) {
+        std::cout << val << ", ";
+      }
+      std::cout << std::endl;
+      std::cout << std::endl;
+#endif
+      intermediate_twiddles_offset += 2 * kernel0.batch_size * static_cast<IdxGlobal>(kernel0.length);
+      impl_twiddle_offset += detail::increment_twiddle_offset(kernel0.level, static_cast<Idx>(kernel0.length));
+      for (std::size_t factor_num = 1; factor_num < static_cast<std::size_t>(dimension0.num_factors); factor_num++) {
+        auto& current_kernel = kernels.at(factor_num);
+        if (static_cast<Idx>(factor_num) == dimension0.num_factors - 1) {
```

Comment:
If we are going to dump all the outputs, we might as well dump them after the last factor as well

Reply:
This is already done - logging is done outside of if/else.